Adição de PersistenciaDeSerie.cpp

andrebarbieri opened the pull request.

andrebarbieri committed:
Create PersisnteciaDeSerie.cpp

doug-nicihoka committed:
Adição de bibliotecas a PersistenciaDeArquivo.cpp

doug-nicihoka committed:
Pequeno erro corrigido

Pequeno erro na (antiga) linha 21 corrigido, melhoria no espaçamento, alteração em certos nomes de variáveis e nas mensagens de erro.

doug-nicihoka committed:
Melhorias e correções do método "obter"

Problema resolvido: código antigo tentava criar uma instância de Serie (classe abstrata), que gerava erro. Agora método adiciona pontos à Serie antes de a retornar.

doug-nicihoka committed:
Pequenas alterações no método "inserir"

Linha 119 (nova): adicionado parâmetro para a escrita no final do arquivo. Variáveis não utilizadas removidas.

doug-nicihoka commented on PersisnteciaDeSerie.cpp:
```diff
@@ -0,0 +1,139 @@
+        out << s->getQuantidade() << std::endl;
+        out << s->getNomeDoCanalX() << std::endl;
+        out << s->getNomeDoCanalY() << std::endl;
+        for (std::vector<Ponto*>::iterator it = s->getPontos()->begin(); it != s->getPontos()->end; ++it) {
```
Esta linha está gerando um erro por algum motivo @andrebarbs13 .

doug-nicihoka committed:
Linhas que geram erro comentadas

Linhas que geram erro de compilação no código comentadas para realização de testes.

doug-nicihoka commented:
@andrebarbs13 testar construtor da classe, pois parece não estar funcionando como esperado no enunciado. Lembre-se que o construtor **não** lança exceção se o arquivo não existe, ou se o arquivo está vazio. Também é bom lembrar que a última linha do arquivo **sempre** estará em branco (talvez ela cause algum problema).

doug-nicihoka committed:
Remoção de comentário, problema encontrado

Erro de compilação ocorria devido à falta de parênteses após a chamada do método "end()" do container vector.

andrebarbieri committed:
Update PersisnteciaDeSerie.cpp

doug-nicihoka committed:
Reversão do commit anterior

doug-nicihoka committed:
Correção do construtor

Construtor estava lançando erro anteriormente, pois lia linha em branco. Parece funcionar para arquivos vazios, arquivos com formato correto, e para arquivos não existentes.

doug-nicihoka committed:
Correção do método obter

Sem essa alteração, o loop inicia-se uma vez a mais do que o necessário (por causa da linha em branco no final do arquivo), o que não causou erros nos testes (mas pode causar algum erro, eventualmente).
